apps/governance/src/components/telemetry-dialog/telemetry-dialog.tsx

```diff
@@ -0,0 +1,129 @@
+import { create } from 'zustand';
+import { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useLocalStorage } from '@vegaprotocol/react-helpers';
+import { Dialog, Icon, Button } from '@vegaprotocol/ui-toolkit';
+import { Networks, useEnvironment } from '@vegaprotocol/environment';
+
+type TelemetryDialogState = {
+  isOpen: boolean;
+  open: () => void;
+  close: () => void;
+};
+
+const useTelemetryDialogStore = create<TelemetryDialogState>((set) => ({
+  isOpen: false,
+  open: () => set({ isOpen: true }),
+  close: () => set({ isOpen: false }),
+}));
+
+export const TELEMETRY_DIALOG_PREVIOUSLY_OPENED =
+  'telemetry_dialog_previously_opened';
+export const TELEMETRY_ON = 'telemetry_on';
+
+export const useTelemetryDialog = () => {
+  const [getPreviouslyOpened, setPreviouslyOpened] = useLocalStorage(
+    TELEMETRY_DIALOG_PREVIOUSLY_OPENED
+  );
+  const [getTelemetryOn, setTelemetryOn] = useLocalStorage(TELEMETRY_ON);
```

Comment:
suggestion (non blocking): `getTelemetryOn` isn't really a function and would be more clearly named just `telemetryOn`